jax/_src/config.py

```diff
@@ -420,7 +420,7 @@ def define_enum_state(
     A contextmanager to control the thread-local state value.
   """
   if not isinstance(default, str):
-    raise TypeError(f"Default value must be of type str, got {default}")
+    raise TypeError(f"Default value must be of type str, got {type(default)}")
```

Comment:
Can we keep the value and add the type? More information is better than less information

Reply:
Sure, I will update them👍🏻.

Reply:
Done 👍🏻, Can you check now?

Reply:
Thanks! I triggered the tests